blog/content/post/2024-11-26-measured-boot-improvements.md

```diff
@@ -0,0 +1,222 @@
+event log. Now the event log format can be selected during build time between:
+
+- coreboot-specific format
+- TCG TPM1.2 format
+- TCG TPM2.0 format
+
+The list of the patches can be found
+[here](https://review.coreboot.org/q/topic:tpm-std-logs). A short
+statistic of total lines of code changed:
+
+- 360 deletions
+- 1675 additions
+
+Having the proper TPM event logging in coreboot made it easier to be parsed
+and used by the coreboot payloads.
+
+## TPM event log from coreboot to payload
+
+### Problem statement
+
+coreboot alone does not create a usable solution. It focuses merely on proper
+silicon initialization. The functionality to boot an OS is left to the
+application called payload. Payload is a piece of software coreboot hands over
+the execution to, after it is done with all its jobs to initialize the silicon
+and platform. It may take many forms:
+
+- [SeaBIOS](https://www.seabios.org/SeaBIOS) - for legacy BIOS compatibility
+- Linux payload/LinuxBoot/heads - a Linux kernel with optional initrd launched
+ directly by coreboot
+- EDKII UEFI Payload - a payload based on [TianoCore
+ EDKII](https://github.com/tianocore/edk2) providing UEFI compatibility
+- [u-boot](https://github.com/u-boot/u-boot) - a bootloader more commonly used
+ on ARM and embedded systems
+- [GRUB2](https://www.gnu.org/software/grub/) - most common bootloader for
+ Linux systems and not only Linux
+- almost any ELF application
+
+The flexibility in payload choice, however, brings certain problems. There are
+still some disputes about the responsibility of coreboot and the
+responsibility of payload. Specific actions made by coreboot may prevent the
+payload from properly implementing particular features, e.g., locking the
+chipset. Sometimes, one may want to skip locking the chipset to allow the
+firmware to be updated by the payload (for example, see
+[heads](https://github.com/linuxboot/heads/pull/1818)). There are more
+situations where some kind of interface between the coreboot and the payload
+must be established. Passing the TPM event log is not different in that case
+but with a slight difference - the event log alone is just a chunk of memory
+that the payload can read and parse however they like. Still, each payload
+would have to do the same: interpret coreboot's custom event log to create a
+single and complete event log from the reset vector to the OS. The amount of
+code duplication it would create is unthinkable. Thankfully, we have already
+implemented the TCG standardized event logging, which helps a bit with the
+portability of the parsing.
+
+In this article, I will show how we achieved the complete integrity of the TPM
+event log across the whole boot process of open-source firmware based on
+coreboot and TianoCore EDKII UEFI Payload in Dasharo.
+
+### Solution
+
+We already have many pieces prepared earlier, like unified TPM support and a
+TCG-compliant event log. The work required to pass the vent log from coreboot
```

Comment:
Spellchecker

Reply:
I put it through grammarly, but vent is not a bad spelled word. It just doesn't fit the context :)

Will change to "event" of course.

Reply:
ff3b5715948bb278461894321745458410058a69